yarn-project/acir-simulator/src/avm/opcodes/external_calls.test.ts

```diff
@@ -0,0 +1,126 @@
+import { Fr } from '@aztec/foundation/fields';
+
+import { jest } from '@jest/globals';
+import { MockProxy, mock } from 'jest-mock-extended';
+
+import { CommitmentsDB, PublicContractsDB, PublicStateDB } from '../../index.js';
+import { AvmMachineState } from '../avm_machine_state.js';
+import { initExecutionEnvironment } from '../fixtures/index.js';
+import { HostStorage } from '../journal/host_storage.js';
+import { AvmJournal } from '../journal/journal.js';
+import { encodeToBytecode } from './encode_to_bytecode.js';
+import { Call } from './external_calls.js';
+import { Opcode } from './opcodes.js';
+
+describe('External Calls', () => {
+  let machineState: AvmMachineState;
+  let journal: AvmJournal;
+
+  let contractsDb: MockProxy<PublicContractsDB>;
+
+  beforeEach(() => {
+    machineState = new AvmMachineState([], initExecutionEnvironment());
+
+    contractsDb = mock<PublicContractsDB>();
+
+    const commitmentsDb = mock<CommitmentsDB>();
+    const publicStateDb = mock<PublicStateDB>();
+    const hostStorage = new HostStorage(publicStateDb, contractsDb, commitmentsDb);
+    journal = new AvmJournal(hostStorage);
+  });
+
+  describe('Call', () => {
+    // TODO(https://github.com/AztecProtocol/aztec-packages/issues/3992): gas not implemented
+    it('Should execute a call correctly', async () => {
+      const gasOffset = 0;
+      const gas = Fr.zero();
+
+      const addrOffset = 1;
+      const addr = new Fr(123456n);
+
+      const argsOffset = 2;
+      const args = [new Fr(1n), new Fr(2n), new Fr(3n)];
+      const argsSize = args.length;
+
+      const retOffset = 8;
+      const retSize = 2;
+
+      const successOffset = 7;
+
+      machineState.writeMemory(0, gas);
+      machineState.writeMemory(1, addr);
+      machineState.writeMemoryChunk(2, args);
+
+      const otherContextInstructions: [Opcode, any[]][] = [
```

Comment:
Encoding like this will not work mid/long term. I'd expect an `encode_bytecode` that takes `Instruction[]` to be more maintainable (but then `Instruction` needs a different design).

Reply:
I agree, i was following a pattern laid out in another pr. I will alter this in future work and not this exact pr